# 4.组学大数据分析


**Next-Generation Sequencing (NGS) data **



###I. 上机指南

---


####0. Data preparation

The Yeast RNA-seq data were downloaded from GSE42983, random sample 100K or 10K reads per sample,  

•	wild-type samples:  two replicates (wt1.fq and wt2.fq)
    
•	H2O2 treatment (oxidative stress) samples: two replicates (wt1X.fq and wt2X.fq)


>**文件位置**：/home/cs/Bioinfo_Lab/3.Genomics/Diff_Express/Raw_reads_10k
>
> You can also download the files by click [here](https://www.jianguoyun.com/p/DTCJc-gQ0NLuBRj9kQ4#dir=%2F2016%2F%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E5%AD%A6%E5%AF%BC%E8%AE%BA%2F%E4%B8%8A%E6%9C%BA%E6%96%87%E4%BB%B6%2F4.Genomics::mode=0).




**Prepare bin environment:**
  
```
安装R软件包

网络学堂下载相应文件，并放到/home/cs/Bioinfo_Lab/3.Genomics/bin/目录下（可能已下载）

cd /home/cs/Bioinfo_Lab/3.Genomics/bin/

tar –zxvf bin.tar.gz （根据实际下载下来的文件名解压，若已下载好则不用解压）

source bin_path_R

cd ..

cd Diff_Express
```

####1. Align the RNA-seq reads to the genome

**1.1 map reads using Tophat **

TopHat maps short sequences from spliced transcripts to whole genomes.

语法：``
tophat [options]  <bowtie_index> <reads1[,reads2,...]> [reads1[,reads2,...]] \[quals1,[quals2,...]] [quals1[,quals2,...]］
``

命令：

``tophat -p 4 -G yeast_annotation.gff --no-coverage-search -o wt1_thout bowtie_index/YeastGenome Raw_reads_10k/wt1.fq 
``

``
tophat -p 4 -G yeast_annotation.gff --no-coverage-search -o wt2_thout bowtie_index/YeastGenome Raw_reads_10k/wt2.fq 
``

``
tophat -p 4 -G yeast_annotation.gff --no-coverage-search -o wt1X_thout bowtie_index/YeastGenome Raw_reads_10k/wt1X.fq 
``

``
tophat -p 4 -G yeast_annotation.gff --no-coverage-search -o wt2X_thout  bowtie_index/YeastGenome Raw_reads_10k/wt2X.fq 
``

```
note：
-p/--num-threads <int>      [ default: 1 ]
-G/--GTF <filename>         (GTF/GFF with known transcripts)
-o/--output-dir <string>    [ default: ./tophat_out ]

```

**1.2 Extract mapped reads on chr I  only (for quick running)**

**index bam file **
```
bamtools index -in wt1_thout/accepted_hits.bam 

bamtools index -in wt2_thout/accepted_hits.bam 

bamtools index -in wt1X_thout/accepted_hits.bam

bamtools index -in wt2X_thout/accepted_hits.bam 
```

>index：Generates index for BAM file

**extract**

```
bamtools filter -region chrI -in wt1_thout/accepted_hits.bam -out wt1_thout/chrI.bam

bamtools filter -region chrI -in wt2_thout/accepted_hits.bam -out wt2_thout/chrI.bam

bamtools filter -region chrI -in wt1X_thout/accepted_hits.bam -out wt1X_thout/chrI.bam

bamtools filter -region chrI -in wt2X_thout/accepted_hits.bam -out wt2X_thout/chrI.bam
```
>filter: Filters BAM files by user-specified criteria





####2. Assemble transcripts on Chr I by Cufflinks 

**2.1 Assemble transcripts for each sample:**
```
cufflinks -p 4 -o wt1_clout wt1_thout/chrI.bam 

cufflinks -p 4 -o wt2_clout wt2_thout/chrI.bam 

cufflinks -p 4 -o wt1X_clout wt1X_thout/chrI.bam 

cufflinks -p 4 -o wt2X_clout wt2X_thout/chrI.bam 
```

**2.2 Create a file called assemblies.txt that lists the assembly files for each sample. **

The file should contain the following lines:

```
wt1X_clout/transcripts.gtf

wt1_clout/transcripts.gtf

wt2X_clout/transcripts.gtf

wt2_clout/transcripts.gtf
```

You can use vi to create a file containing this command:

```
rm -f assemblies.txt;
for i in `ls | grep clout`;
do 
  echo ./$i/transcripts.gtf >> assemblies.txt;
done
```


**2.3 Merge all assemblies to one file containing merged transcripts: **

cuffmerge takes two or more Cufflinks GTF files and merges them into a single unified transcript catalog

``
cuffmerge -g yeast_chrI_annotation.gff -s bowtie_index/YeastGenome.fa  -p 4 assemblies.txt  
``



####3. Identify differentially expressed genes and transcripts 
(we’ll show the output from 1M_reads, not 10K_reads)

Run cuffdiff based on the merged transcripts (assemblies.txt) and mapped reads for each sample (*.bam)

``
cuffdiff -o diff_out -b bowtie_index/YeastGenome.fa -p 4 -u merged_asm/merged.gtf ./wt1_thout/chrI.bam, ./wt2_thout/chrI.bam   ./wt1X_thout/chrI.bam, ./wt2X_thout/chrI.bam  
``

(This won’t generate sufficient result from 10K reads, see diff_out of 1M reads in examples/plot_example_1Mreads/diff_out diff_out)




####4. Explore differential analysis results with CummeRbund TIMING variable
(we’ll use the output from 1M_reads, not 10k reads)

**requirements**

Cufflinks >= v2.0.0

SQLite

R >= v2.7.0

Required Packages:

    RSQLite
    ggplot2 v0.9.2
    reshape2
    plyr
    fastcluster
    rtracklayer
    Gviz
    BiocGenerics (>=0.3.2)
Recommended Packages:     

    Hmisc


命令

``
Rscript  plot_DE_chart.R
``

(See example figures of 1M reads in examples/plot_example_1Mreads/DE_plots)


###II.上机任务

---
**1. 理解和掌握基本的基因差异表达分析 （Differential Expression Analysis）方法; 学会Differential Expression Analysis的基本软件。**

**2.按照Instruction上的方法，使用 TopHat 和Cufflinks完成Differential Expression Analysis。 
**

###III.上机作业

---
**1. 阅读： Trapnell C  et al.    Nat Protoc. 2012 Mar 1;7(3):562-78   Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks.**

[http://www.ncbi.nlm.nih.gov/pubmed/22383036](http://www.ncbi.nlm.nih.gov/pubmed/22383036)

**2. 画一张Plot展示differentially expressed transcripts, 并写一段话解释Plot**

**3. 寻找differentially expressed transcripts，要对数据做怎样的处理？这些处理有哪些统计学上的考虑？**




